scripts/publish_benchmark_result.py

```diff
@@ -0,0 +1,414 @@
+        results_dict,
+        orient="index",
+        columns=columns,
+    )
+
+    print("---BIGQUERY USAGE REPORT---")
+    for index, row in benchmark_metrics.iterrows():
+        print(
+            f"{index} - query count: {row['Query_Count']},"
+            f" bytes processed sum: {row['Bytes_Processed']},"
+            f" slot millis sum: {row['Slot_Millis']},"
+            f" local execution time: {row['Local_Execution_Time_Sec']} seconds"
+            f" bigquery execution time: {row['BigQuery_Execution_Time_Sec']} seconds"
+        )
+
+    cumulative_queries = benchmark_metrics["Query_Count"].sum()
+    cumulative_bytes = benchmark_metrics["Bytes_Processed"].sum()
+    cumulative_slot_millis = benchmark_metrics["Slot_Millis"].sum()
+    cumulative_local_seconds = benchmark_metrics["Local_Execution_Time_Sec"].sum(
+        skipna=True
+    )
+    cumulative_bq_seconds = benchmark_metrics["BigQuery_Execution_Time_Sec"].sum()
+
+    print(
+        f"---total queries: {cumulative_queries}, "
+        f"total bytes: {cumulative_bytes}, "
+        f"total slot millis: {cumulative_slot_millis}, "
+        f"Total local execution time: {cumulative_local_seconds} seconds, "
+        f"Total bigquery execution time: {cumulative_bq_seconds} seconds---"
+    )
+
+    return benchmark_metrics.reset_index().rename(columns={"index": "Benchmark_Name"})
+
+
+def get_repository_status():
+    git_hash = subprocess.check_output(
+        ["git", "rev-parse", "--short", "HEAD"], text=True
+    ).strip()
+    bigframes_version = subprocess.check_output(
+        ["python", "-c", "import bigframes; print(bigframes.__version__)"], text=True
+    ).strip()
+    release_version = (
+        f"{bigframes_version}dev{datetime.datetime.now().strftime('%Y%m%d')}+{git_hash}"
+    )
+
+    return {
+        "benchmark_start_time": datetime.datetime.now().isoformat(),
+        "git_hash": git_hash,
+        "bigframes_version": bigframes_version,
+        "release_version": release_version,
+        "python_version": sys.version,
+    }
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Run benchmarks for different scenarios."
+    )
+    parser.add_argument(
+        "--notebook",
+        action="store_true",
+        help="Run benchmarks associated with notebooks",
+    )
+    args = parser.parse_args()
+
+    is_kokoro = "KOKORO_JOB_NAME" in os.environ
+
+    if is_kokoro:
+        current_directory = os.getcwd()
+        subprocess.run(
+            ["git", "config", "--global", "--add", "safe.directory", current_directory],
+            check=True,
+        )
+
+    repo_status = get_repository_status()
+    if args.notebook:
+        bigquery_table = "bigframes-metrics.benchmark_report.notebook_benchmark"
+        benchmark_metrics = run_notebook_benchmark()
+    else:
+        bigquery_table = "bigframes-metrics.benchmark_report.benchmark"
+        benchmark_metrics = run_benchmark(Path("tests/benchmark/"))
```

Comment:
For both `run_notebook_benchmark()` and `run_benchmark(...)`, we need to check if any subprocess has failed (non-zero return code) and make sure this script exits with a non-zero return code if so.

Reply:
Updated to not use subprocess to capture issue.